hivemind/utils/asyncio.py

```diff
@@ -59,6 +59,16 @@ async def aenumerate(aiterable: AsyncIterable[T]) -> AsyncIterable[Tuple[int, T]
         index += 1
 
 
+async def asingle(aiter: AsyncIterable[T]) -> T:
```

Comment:
The name is inspired by [Single()](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.single?view=net-5.0) from LINQ (.NET functional programming functions).